mwana/apps/export/views.py

```diff
@@ -0,0 +1,132 @@
+# vim: ai ts=4 sts=4 et sw=4
+# Create your views here.
+import psycopg2
+from mwana import localsettings as settings
+from django.http import HttpResponse
+import json
+from mwana.apps.locations.models import Location
+from collections import OrderedDict
+
+
+def get_turnaround(request):
+    '''
+    key_value = ""
+    if ((request.method == 'GET') and ('key_value' in request.GET)):
+        key_value = request.GET['key_value']
+    '''
+    connection = psycopg2.connect(host=settings.DATABASES['default']['HOST'], database=settings.DATABASES['default']['NAME'],\
+                    user=settings.DATABASES['default']['USER'], password=settings.DATABASES['default']['PASSWORD'])
+    
+    districts = Location.objects.filter().distinct()
+    
+    site_info_req = OrderedDict()
+
+    try:
+        cursor = connection.cursor()
+        cursor.execute(SQL_QUERY_SELECT) 
+        site_info_req = dictfetchall(cursor)
+
+    except:
+        log = "oops...", sys.exc_info()
+    finally:
+        connection.close()
+        return HttpResponse(json.dumps(site_info_req))   
+
+def dictfetchall(cursor):
+    # "Returns all rows from a cursor as a dict"
+    to_return_dict = []
+    desc = cursor.description
+    if cursor.rowcount > 0:
+        o_dict_rows = [
+            OrderedDict(zip([str(col[0]) for col in desc], row))
+            for row in cursor.fetchall()
+        ]
+
+        for row in o_dict_rows:
+            o_dict_str = OrderedDict()
+            for key,val in row.items():
+                o_dict_str[key] = str(val)
+
+            to_return_dict.append(o_dict_str)
+    else:
+        cols = [desc[0] for desc in cursor.description]
+        dict_cols = OrderedDict()
+        for col in cols:
+            dict_cols[col] = ''
+        to_return_dict.append(dict_cols)
+
+    return to_return_dict
+
+SQL_QUERY_SELECT = '''
```

Comment:
Hi Trevor,

Paul tells me you resolved this when you met?

When can we get a merge?

On Mon, Feb 1, 2016, 12:19 PM Trevor Sinkala <notifications@github.com>
wrote:

> In mwana/apps/export/views.py
> <https://github.com/mwana/mwana/pull/192#discussion_r51400327>:
>
> > +        for row in o_dict_rows:
> > +            o_dict_str = OrderedDict()
> > +            for key,val in row.items():
> > +                o_dict_str[key] = str(val)
> > +
> > +            to_return_dict.append(o_dict_str)
> > +    else:
> > +        cols = [desc[0] for desc in cursor.description]
> > +        dict_cols = OrderedDict()
> > +        for col in cols:
> > +            dict_cols[col] = ''
> > +        to_return_dict.append(dict_cols)
> > +
> > +    return to_return_dict
> > +
> > +SQL_QUERY_SELECT = '''
>
> Hi Paul, In the case where you have even number of records, say (11, 22,
> 33, 44) are we getting 22 or 33 or (22 + 33)/2? Thank you, Trevor
> … <#msg-f:1524967100446365667_>
> On Feb 1, 2016 12:00 PM, "Andre Lesa" <notifications@github.com> wrote:
> In mwana/apps/export/views.py <#192 (comment)
> <https://github.com/mwana/mwana/pull/192#discussion_r51398417>>: > + for
> row in o_dict_rows: > + o_dict_str = OrderedDict() > + for key,val in
> row.items(): > + o_dict_str[key] = str(val) > + > +
> to_return_dict.append(o_dict_str) > + else: > + cols = [desc[0] for desc in
> cursor.description] > + dict_cols = OrderedDict() > + for col in cols: > +
> dict_cols[col] = '' > + to_return_dict.append(dict_cols) > + > + return
> to_return_dict > + > +SQL_QUERY_SELECT = ''' Hi Trevor, If you look through
> the SQL, you’ll observe we used the median for the turnaround as you
> previously advised and not the mean. Regards, Paul C. Kaumba Software
> Developer T +260977664404 M +260966664404 Skype paul.kaumba-palladium
> Catholic Relief Services (CRS) Palladium Make it possible
> thepalladiumgroup.com<http://thepalladiumgroup.com/> From: Andre Lesa
> [mailto:andre@andrelesa.com] Sent: Monday, February 01, 2016 5:37 AM To:
> mwana/mwana <reply@reply.github.com>; mwana/mwana <
> mwana@noreply.github.com>; Kaumba, Paul <Paul.Kaumba@thepalladiumgroup.com>
> Subject: Re: [mwana] Created PMTCT Dashboard mwana data exporter (#192
> <https://github.com/mwana/mwana/pull/192> <#192
> <https://github.com/mwana/mwana/pull/192>) Thanks Trevor, Looping in Paul
> who wrote the queries. On Mon, Feb 1, 2016, 12:08 AM Trevor Sinkala <
> notifications@github.com<mailto:notifications@github.com>> wrote: In
> mwana/apps/export/views.py<#192 <https://github.com/mwana/mwana/pull/192>
> (comment) <#192 (comment)
> <https://github.com/mwana/mwana/pull/192#discussion_r51368442>>>: + for
> row in o_dict_rows: + o_dict_str = OrderedDict() + for key,val in
> row.items(): + o_dict_str[key] = str(val) + +
> to_return_dict.append(o_dict_str) + else: + cols = [desc[0] for desc in
> cursor.description] + dict_cols = OrderedDict() + for col in cols: +
> dict_cols[col] = '' + to_return_dict.append(dict_cols) + + return
> to_return_dict + +SQL_QUERY_SELECT = ''' Query seems to run fast, but can
> we use true median? In some cases the differences are quite significant if
> you use median — Reply to this email directly or view it on GitHub<
> https://github.com/mwana/mwana/pull/192/files#r51368442>. — Reply to this
> email directly or view it on GitHub <
> https://github.com/mwana/mwana/pull/192/files#r51398417>.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mwana/mwana/pull/192/files#r51400327>.
>